python/triton/runtime/autotuner.py

```diff
@@ -102,14 +103,18 @@ def kernel_call():
             if config.pre_hook:
                 config.pre_hook(full_nargs)
             self.pre_hook(args)
-            self.fn.run(
-                *args,
-                num_warps=config.num_warps,
-                num_stages=config.num_stages,
-                num_ctas=config.num_ctas,
-                **current,
-            )
-            self.post_hook(args)
+            try:
+                self.fn.run(
+                    *args,
+                    num_warps=config.num_warps,
+                    num_stages=config.num_stages,
+                    num_ctas=config.num_ctas,
+                    **current,
+                )
+            except OutOfResources as e:
+                raise e
+            finally:
+                self.post_hook(args, from_exception=True)
```

Comment:
The commit message says we expose the exception to the user for inspection, which I assumed meant we passed it to the post_hook, but we don't seem to do that?  I don't really care whether or not we do, but should make the commit message match the code.

Reply:
I'll change the commit message. It's not exposing the exception itself, but instead just whether `post_hook` is invoked when exception is raised.